odoo/modules/registry.py

```diff
@@ -555,6 +555,7 @@ def check_tables_exist(self, cr):
             # recreate missing tables
             for name in missing:
                 _logger.info("Recreate table of model %s.", name)
+                env[name]._auto_init()
                 env[name].init()
             env.flush_all()
```

Comment:
This code iterates through the models that have no table in the database.  It makes the assumption that those models are SQL views which are created by invoking the method `init()`.  The method `_auto_init()` is not supposed to be invoked here.